pandas/tests/io/parser/test_parse_dates.py

```diff
@@ -1290,16 +1290,27 @@ def test_parse_dates_empty_string(all_parsers):
     tm.assert_frame_equal(result, expected)
 
 
-def test_parse_dates_infer_datetime_format_warning(all_parsers):
-    # GH 49024
+@pytest.mark.parametrize(
+    "reader",
+    [
+        'read_csv_check_warnings',
+        'read_table_check_warnings'
+    ]
+)
+def test_parse_dates_infer_datetime_format_warning(all_parsers, reader):
+    # GH 49024, 51017
     parser = all_parsers
     data = "Date,test\n2012-01-01,1\n,2"
-    parser.read_csv_check_warnings(
-        UserWarning,
+    warning_type = (FutureWarning if reader == "read_table_check_warnings"
+                    else UserWarning)
```

Comment:
these should both be `FutureWarning` - looks like the read_csv one was wrong to begin with, could you change that to be `FutureWarning` too please?